seannkelleyy/Mario-Remake
# IH 78 mario death handling

﻿
# Description

New feat: Added lives to Mario. When the player dies, Mario loses one life and the game restarts with Mario's new number of lives. 
New feat: Added GameRestarter singleton because there are now multiple places where the game can be restarted. 
New feat: Added a way to edit the JSON file when Mario loses a life and when the game exits to reset Mario's number of lives back to 10.

closes #78

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How To Test

Run the game and make Mario die. Then check the JSON file to make sure Mario's lives decreased by 1. Then run the game again and press "Q" to quit the game. Check the JSON again to make sure Mario's lives were reset to 10. 

# Checklist:
